payroll/models/hr_payslip.py

```diff
@@ -362,11 +265,21 @@ def compute_sheet(self):
                     date_from=payslip.date_from, date_to=payslip.date_to
                 ).ids
             )
-            lines = [
-                (0, 0, line)
-                for line in self._get_payslip_lines(contract_ids, payslip.id)
-            ]
-            payslip.write({"line_ids": lines, "number": number})
+            # get localdict and linesdict
+            localdict, lines_dict = self._get_payslip_lines(contract_ids, payslip.id)
```

Comment:
The point of returning localdict is that when you inherit this method, you can't for example add a functionality that requires localdict because the function always return the lines. 

For example this is useful if you have another type of categories in a private module and you need to don some method like _sum_by_salary_rule_category. 

With the current implementation you can't access localdict inheriting any function and that limits the developer to interact and change available values in localdict. 

It's no used because it's meant to be inherited, for example Can be used in @appstogrow PR or in any implementation requiring to modify localdict on the middle of calculation. 

Hope it's clear if not I can submit que example in the code.